web/src/app/modules/shared/components/presentation/modal/modal.component.html

```diff
@@ -0,0 +1,122 @@
+                <label>{{ opt.label }}</label>
+              </clr-radio-wrapper>
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-radio-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'text'">
+            <clr-input-container>
+              <label [for]="field.name">{{ field.label }}</label>
+              <input
+                clrInput
+                type="text"
+                [name]="field.name"
+                [formControlName]="field.name"
+                [placeholder]="field.placeholder"
+              />
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-input-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'number'">
+            <clr-input-container>
+              <label [for]="field.name">{{ field.label }}</label>
+              <input clrInput type="number" [formControlName]="field.name" />
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-input-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'password'">
+            <clr-input-container>
+              <label [for]="field.name">{{ field.label }}</label>
+              <input
+                clrInput
+                type="password"
+                [formControlName]="field.name"
+              />
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-input-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'select'">
+            <clr-select-container>
+              <label [for]="field.name">{{ field.label }}</label>
+              <select
+                clrSelect
+                [formControlName]="field.name"
+                [multiple]="field.configuration.multiple"
+              >
+                <option
+                  *ngFor="let opt of fieldChoices(field); trackBy: trackByFn"
+                  [value]="opt.value"
+                >
+                  {{ opt.label }}
+                </option>
+              </select>
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-select-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'textarea'">
+            <clr-textarea-container>
+              <label [for]="field.name">{{ field.label }}</label>
+              <textarea clrTextarea [formControlName]="field.name"></textarea>
+              <clr-control-error>{{ field.error }}</clr-control-error>
+            </clr-textarea-container>
+          </ng-container>
+          <ng-container *ngSwitchCase="'hidden'"> </ng-container>
+          <ng-container *ngSwitchDefault>
+            Unable to display form field type {{ field.type }}
+          </ng-container>
+        </ng-container>
+        <br>
+      </div>
+    </form>
+  </div>
+  <div class="modal-footer">
+    <ng-container *ngIf="form else closeButton">
```

Comment:
Would the modal component be more useful if it gave more choice to the user? For example, let the user decide which buttons they want to show and what should happen if they show those buttons. For code that creates a modal component, create helpers that make common tasks trivial.

Reply:
The modal now supports three cases of usage. Forms still have a close or submit buttons. For all other content, the user can either choose to provide button(s) or omit adding any buttons entirely where a close button will fill instead.